persistence/sql/persister_consent.go

```diff
@@ -214,146 +211,186 @@ func (p *Persister) GetConsentRequest(ctx context.Context, challenge string) (*c
 		return nil, err
 	}
 
+	// We need to overwrite the ID with the encoded flow (challenge) so that the client is not confused.
+	f.ConsentChallengeID = sqlxx.NullString(challenge)
```

Comment:
Can you explain this a bit more?

Reply:
We check that the returned consent request challenge ID matches the passed consent challenge here: https://github.com/ory/hydra/blob/012bb3543dc3ac2a15fae517f462d4c57df22c2a/oauth2/oauth2_auth_code_test.go#L173

Reply:
This ensures that the challenge ID is set correctly in the returned (partial) flow. Since the consent challenge in the flow is an encoding of the flow itself, we can't set that field in the flow beforehand.